build.gradle.kts

```diff
@@ -138,15 +140,35 @@ allprojects {
             }
         }
 
-        // Configure existing tasks
+        build {
+            dependsOn(updateLicenses)
+            finalizedBy(findByName("copyArtifacts"), clean)
+        }
+
         test {
             useJUnitPlatform()
             val maxWorkerCount = gradle.startParameter.maxWorkerCount
             maxParallelForks = if (maxWorkerCount < 2) 1 else maxWorkerCount / 2
+            finalizedBy(jacocoTestReport) // report is always generated after tests run
         }
-        build {
-            dependsOn(updateLicenses)
-            finalizedBy(findByName("copyArtifacts"), clean)
+
+        jacocoTestReport {
+            reports {
+                csv.required.set(false)
+                xml.required.set(true)
+                html.required.set(true)
+            }
+        }
+
+        jacocoTestCoverageVerification {
+            dependsOn(jacocoTestReport) // report is required to run before verification
+            violationRules {
+                rule {
+                    limit {
+                        minimum = "0.8".toBigDecimal()
```

Comment:
Limitation of JaCoCo configuration. This _has_ to be a BigDecimal. Instead of importing, this is the recommended solution